### PR TITLE
Set GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH to /usr/include/<library arch> only if it exists.

### PR DESCRIPTION
**Pull request by plotfi**
When enabling runtimes builds to build libcxx I am running into an issue where GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH is set to /usr/include/<library arch> even when this location doesn't change. I did some git commit archaeology and found that there originally was a fixme comment on this line about how this behavior affects cross compiling (which can probably also apply with runtimes builds too). 

So I added the comment back, and did the expedient thing for now: set GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH to include CMAKE_LIBRARY_ARCHITECTURE as part of the /usr/include path only if it actually exists.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @Rostepher @gottesmm @drodriguez

File: stdlib/public/Platform/CMakeLists.txt [hunk 1]
```diff
@@ -84,7 +84,19 @@ foreach(sdk ${SWIFT_SDKS})
...
+      #        machine's path, regardless of the SDK target we're building for.
+      #        This will break if cross-compiling from a distro that uses the
+      #        architecture as part of the path to a distro that does not.
+      #
+      #        However, this is not always the case. As a stopgap solution, we
+      #        will set the GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH depending
+      #        on if the arch_include_path/library_arch path exists.
+      if(EXISTS "${GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH}/${CMAKE_LIBRARY_ARCHITECTURE}")
+        set(GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH "${GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH}/${CMAKE_LIBRARY_ARCHITECTURE}")
+      endif()
```
Comment (drodriguez) [inline stdlib/public/Platform/CMakeLists.txt]: If I'm seeing this correctly, `GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH` ends up being passed to two GYB invocations below. If this value is not set, those invocations will have an empty string for that value and the modulemap will be incorrect. Or am I missing something?

I might be wrong, but I think `CMAKE_LIBRARY_ARCHITECTURE` is empty for systems that do not use it, so the previous version is basically like setting `GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH` to the previous value with and extra trailing separator.

Comment (plotfi) [reply stdlib/public/Platform/CMakeLists.txt]: @drodriguez For runtimes libcxx builds it gets set to the runtime target name.

This is a check to see if the system has a directory like /usr/include/linux-x86_64/ (ie ${GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH}/${CMAKE_LIBRARY_ARCHITECTURE}). Exists or not, and if not then do not append CMAKE_LIBRARY_ARCHITECTURE to the path. 

In this case the CMAKE_LIBRARY_ARCHITECTURE is not empty; it's set to the target name for the runtime.

Comment (plotfi) [reply stdlib/public/Platform/CMakeLists.txt]: I will try and set the CMAKE_LIBRARY_ARCHITECTURE to empty in cmake cache files and see what happens. I still think the comment should probably remain.

Comment (plotfi) [reply stdlib/public/Platform/CMakeLists.txt]: @drodriguez Still trying to understand how swift stdlib is generating the CMakeCCompiler.cmake/CMakeCXXCompiler.cmake files that appear to set CMAKE_LIBRARY_ARCHITECTURE.

Comment (drodriguez) [reply stdlib/public/Platform/CMakeLists.txt]: I wonder if this is something with our CentOS. Or did you experience this in other OS? Maybe tomorrow we can check this in a Ubuntu machine. I believe what you are telling me, and if the value of `CMAKE_LIBRARY_ARCHITECTURE` is not empty, the fix is correct, but I wonder what's happening here.

Comment (plotfi) [reply stdlib/public/Platform/CMakeLists.txt]: @drodriguez @compnerd I've updated this patch to just provide a way to have an override path to GLIBC_SYSROOT_RELATIVE.*_INCLUDE_PATH. What do you all think?

**Comment by compnerd** [comment]
I think that we should be doing this per target since you can be building multiple targets at once.  Furthermore, I think that this can be done just inline since we will have the configuration defaulted.

**Pull request closed without merging**